_posts/2024-08-23-gsoc24-final-report.md

```diff
@@ -0,0 +1,133 @@
+
+2. [Introduction to Integration Testing in kworkflow ]({{site.url}}/introduction-to-integration-testing/)
+
+3. [Integration Testing for kw ssh]({{site.url}}/integration-for-kw-ssh/)
+
+4. [Integration Testing for kw build]({{site.url}}/integration-for-kw-build/)
+
+# Contributions
+
+Throughout the project, I created several pull requests (PRs) addressing
+different aspects of kworkflow. Each PR was carefully crafted to enhance
+functionality, increase test coverage, and ensure code robustness. Below are
+some of the most significant PRs:
+
+| Pull Request | N° of Commits | 
+|--------------|:---------------:|
+| [setup: install kernel build dependencies](https://github.com/kworkflow/kworkflow/pull/1108) |4|
+| [tests: integration: device_test: modify kw device integration test to run entirely in container](https://github.com/kworkflow/kworkflow/pull/1135) | 2 |
+| [tests: integration: refactor kw_version_test to run entirely in container](https://github.com/kworkflow/kworkflow/pull/1113) | 1 |
+| [run_tests: dedicate a container per test file for integration tests](https://github.com/kworkflow/kworkflow/pull/1130) | 2 |
+| [run_tests: streamline test execution logic](https://github.com/kworkflow/kworkflow/pull/1148) | 1 |
+| [tests: integration: self_update_test: add self-update test](https://github.com/kworkflow/kworkflow/pull/1055) | 3 |
+| [tests: integration: deploy_test: introducing deploy tests](https://github.com/kworkflow/kworkflow/pull/1161) | 2 |
+| [tests: integration: kw_ssh_test: Add integration tests for kw ssh functionality](https://github.com/kworkflow/kworkflow/pull/1116) | 5 |
+| [tests: integration: build_test: add the kw build test](https://github.com/kworkflow/kworkflow/pull/1143) | 3 |
+
+### Features in Development: Almost Ready to be Merged
+
+The PRs listed above include features that are actively in development, such as
+tests for the `kw ssh`, `kw build`, and `kw deploy` functionalities. Although
+many of these PRs are already well-structured, they are still undergoing final
+reviews and refinements. A significant portion of the important decisions has
+already been discussed within the team, ensuring that the development remains
+aligned with the project’s goals and needs.
+
+# Next Steps
+
+As a long-time contributor to **kworkflow**, I am committed to continuing my
+contributions to the project. Here are the key areas I plan to focus on:
+
+1. **Implementing Acceptance Tests**:
+ I plan to develop acceptance tests that will validate multiple functionalities
+ in sequence. These tests will ensure that the integration of various features
+ works seamlesslyand meets the overall requirements of the project.
+
+2. **Expanding Test Coverage**:
+ Continuing to expand the test coverage is a priority. I will work on creating
+ and refining tests for additional functionalities that have not yet been fully
+ covered, ensuring a comprehensive and effective validation process.
```

Comment:
Add something like this at the end of this bullet:

"However, it is important to note that by covering the `kw build` and `kw deploy` features, which involved the more complex and particular infrastructure, along with features such as `kw ssh`, `kw device`, and the like, expanding the suite for the remaining features will surely be easier and straightforward."

Of course, you don't need to copy and paste exactly what I wrote, but give the reader (who doesn't understand your project as well as you and me) the sense that this initial hard work of covering the "**all** different types of integration tests" pays off as it streamlines the addition and revision of tests.